crates/ty/build.rs

```diff
@@ -12,12 +12,36 @@ fn main() {
         .join("..")
         .join("..");
 
+    version_info(&workspace_root);
     commit_info(&workspace_root);
 
     let target = std::env::var("TARGET").unwrap();
     println!("cargo::rustc-env=RUST_HOST_TARGET={target}");
 }
 
+/// Retrieve the version from the `dist-workspace.toml` file and set `TY_VERSION`.
+fn version_info(workspace_root: &Path) {
+    let dist_file = workspace_root.join("dist-workspace.toml");
```

Comment:
I think you have to check the parent dir actually (and consider it not an error for no such file to exist).

Reply:
wdym?

The parent dir change is in #17866